### ISSUE TITLE
rev721/722 1 MIPS Textures wrongly detected as correct

### ISSUE THREAD

**Issue by starkgate** [ME3Explorer/ME3Explorer#101]
1 MIPS textures aren't flagged as MIPS in TPF/DDS upon analysing, even though selecting them does show that their MIPS number is different from the tree. 
"Forcing" individual autofix by selecting them and clicking autofix on the menu that appears does fix them. Using global autofix doesn't fix them.

**Comment by starkgate** [comment]
Test files : http://filebin.ca/2MJsrd4vXYcC/1MIPS.7z

**Comment by KFreon** [comment]
I debated with myself for many hours over this. The differing mips is no longer a real issue as the toolset issues a request for an image from the ImageEngine either way. 
When it does, it mentions how many mips it wants and the ImageEngine simply generates them. 
tldr: Autofix is ONLY required for format fixing now. 

The interesting thing about that is that technically, nothing ever needs to be "autofixed" for the same reasoning. A request is issued, it mentions what it wants, and gets that back.
In the format case, I decided that the toolset should provide some negative feedback since a different format is a whole new story, whereas mips are just a collection of smaller images in the same format.

I'm open to suggestions as to which way to go. Options are:
1) Ignore this report and take it as a feature.
2) Re-enable mips flagging when different.
3) Remove autofix entirely, and have it occur as required automatically.

Personally, I like the bottom one but I think it should only be implemented once all the bugs are ironed out of the conversion process.

**Comment by starkgate** [comment]
Hmm. Interesting indeed.
I use autofix to fix textures for ALOT (my mod), and I do need to make sure textures are fixed for real, entirely, so that the user has nothing else to do than to install. 2 main reasons for that :
- It's quicker for the user. I do the process 1 time, that saves time for 1000s of people.
- It doesn't rely on any revision requirement. A fixed TPF can be installed on absolutely any rev without problems (except the cursed 718 of course^^). No need of further autofix, dependent of the rev. 

For MIPS in particular, different revs produce different results (eg 12 +- 1 MIPS for a 4k texture). It does induce crashes and graphical glitches, so not a great thing.

The best option here, imo, is to give the user the most control on what he does, ie keep autofix and leave the choice to the user. It's generally annoying to have a tool that does something that you don't expect it to do, without noticing you of it.
That being said, it could absolutely be integrated as an option for advanced users, saving a bit of hassle when installing (only one button to click).

PS : I don't check on github as often as the forum, you can send me a PM when you want me to come check out your answer :).

**Comment by KFreon** [comment]
I'm not 100% on what you mean here. Are you saying you'd prefer to have mips shown as incorrect?
I don't get how it can make different mips for the same size of texture...
Is it format related?

**Comment by starkgate** [comment]
Sorry, I'll try to be more clear this time :

> Are you saying you'd prefer to have mips shown as incorrect?

Yes, I'd prefer that to an automatic fix that doesn't notice you of it until it's too late.

> I don't get how it can make different mips for the same size of texture...

This refers to #75. If such a problem would happen again in a future version, not having control on the autofix could be problematic.

**Comment by KFreon** [comment]
Fair enough. Well I'll see if I can get it to do it again.

**Comment by starkgate** [comment]
Same behaviour in 723 here, which is expected as I don't think you changed anything on this side^^.

**Comment by KFreon** [comment]
Oh whoops. No I didn't :)

**Comment by starkgate** [comment]
Doesn't work in 725. 2 problems :
- The MIPS count is very wrong, TPF/DDS lists for example 12 MIPS in 725 instead of 1 in 718.
- Previews are entirely black for textures without MIPS.
  ![capture](https://cloud.githubusercontent.com/assets/12874678/11994667/e18c9852-aa42-11e5-9310-d3f9fb8e6c77.PNG)

**Comment by KFreon** [comment]
Pretty sure this is fixed now.

**Comment by starkgate** [comment]
It is the exact same behaviour in 726, black previews and wrong MIPS count.

**Comment by KFreon** [comment]
Works for me on Windows 7 :(
![win7 modmaker correct](https://cloud.githubusercontent.com/assets/7845997/12532862/8ac17624-c26a-11e5-97ff-7dcbda4b1753.jpg)

**Comment by starkgate** [comment]
Oh, yes it works since a few revs. I thought I had updated you about that, sorry (maybe it was in another bug report, I don't remember).

**Comment by KFreon** [comment]
Oh wait...my bad.
Now I see you said that 25 days ago :D
